Plugins/Flow.Launcher.Plugin.ProcessKiller/ProcessHelper.cs

```diff
@@ -38,10 +38,19 @@ public List<ProcessResult> GetMatchingProcesses(string searchTerm)
         {
             var processlist = new List<ProcessResult>();
 
+            int portNum;
+            bool canConvert = int.TryParse(searchTerm, out portNum);
+            Tuple<bool, PortDetail> tcpPortListeningProcess = null;
+            if(canConvert)
+                tcpPortListeningProcess = PortHelper.GetPortDetails(portNum);
+
             foreach (var p in Process.GetProcesses())
             {
                 if (IsSystemProcess(p)) continue;
-
+                if (tcpPortListeningProcess != null && tcpPortListeningProcess.Item1 && tcpPortListeningProcess.Item2.Process.Id == p.Id)
```

Comment:
The tuple is not supposed to be null (ValueTuple is not able to be null without explicit written). So maybe you may need to adjust part of this.

Reply:
Check pattern matching if you have time. https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching